### ISSUE TITLE
Support XDG Base Directory Specification

### ISSUE THREAD

**Issue by sagikazarmark** [thought-machine/please#1501]
See https://github.com/thought-machine/please/issues/1253#issuecomment-766426642

- [x] Add `$XDG_CONFIG_HOME` before checking `~/.config/please/plzconfig` #1518
- [ ] Download Please tools to `$XDG_DATA_HOME`

https://wiki.archlinux.org/index.php/XDG_Base_Directory

**Comment by Tatskaari** [comment]
The goal is to make please download tools automatically rather than have them be part of the distribution. These will be part of the build graph so will be downloaded them into `plz-out`. 

Happy to accept a PR to add `$XDG_CONFIG_HOME` to the list of places Please looks for config though.

**Comment by sagikazarmark** [comment]
Downloading the please binary is still an issue, is it not?

**Comment by Tatskaari** [comment]
> Downloading the please binary is still an issue, is it not?

Yeah okay I think this needs a bit of thought. I want to change the way we install Please soon anyway. Symlinking please about has proven to be problematic. I think this is the next bit of work I want to address.

**Comment by sagikazarmark** [comment]
I think it would make sense to use environment variables for please location configuration instead of config file options. For one, it works better with `pleasew` and integrates better into user environments. The XDG standard is a great solution in my opinion.

**Comment by sagikazarmark** [comment]
:(

Can someone reopen please?

**Comment by peterebden** [comment]
Agreed, bad bot!
